ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/part/PartStackPresenter.java

```diff
@@ -145,14 +145,8 @@ public void addPart(@NotNull PartPresenter part) {
     @Override
     public void addPart(@NotNull PartPresenter part, @Nullable Constraints constraint) {
         if (containsPart(part)) {
-            workBenchPartController.setHidden(true);
-
-            TabItem selectedItem = getTabByPart(part);
-
-            if (selectedItem != null) {
-                selectedItem.unSelect();
-            }
-
+            TabItem tab = getTabByPart(part);
+            onTabClicked(tab);
```

Comment:
```
        if (containsPart(part)) {
            TabItem tab = getTabByPart(part);
            onTabClicked(tab);
            return;
        }
```
Yes, the case is confuses a little bit. But here **tab** must be not **null** because of there is a verification before.